tests/test_job_tracker.py

```diff
@@ -0,0 +1,385 @@
+                "job_name": "TestJob1",
+            },
+            "job2": {
+                "start_time": current_time - timedelta(seconds=60),
+                "job_name": None,
+            },
+            "job3": {
+                "start_time": current_time - timedelta(seconds=300),
+                "job_name": "job3",  # Same as ID, should not show in parentheses
+            },
+        }
+
+        job_tracker._handle_sigusr1()
+
+        # Verify logging was called with job information
+        mock_log.info.assert_any_call(f"Total running jobs: {len(job_tracker.running_jobs)}")
+        # Check that the table was printed (header should be present)
+        assert any("Job ID" in str(call) and "Duration" in str(call) for call in mock_log.info.call_args_list)
+
+    def test_on_job_submitted_should_track_new_job(self, mock_scheduler):
+        """Test tracking when a new job starts."""
+        job_tracker = JobTracker()
+        job_tracker.scheduler = mock_scheduler
+
+        # Create mock job with name attribute properly set
+        mock_job = Mock()
+        mock_job.name = "TestJob"
+        mock_scheduler.get_job.return_value = mock_job
+
+        event = Mock(spec=JobSubmissionEvent)
+        event.job_id = "job123"
+        event.scheduled_run_times = [datetime(2024, 1, 1, 10, 0, 0)]
+        event.jobstore = "default"
+
+        job_tracker.on_job_submitted(event)
+
+        assert "job123" in job_tracker.running_jobs
+        assert job_tracker.running_jobs["job123"]["scheduled_run_time"] == datetime(2024, 1, 1, 10, 0, 0)
+        assert job_tracker.running_jobs["job123"]["jobstore"] == "default"
+        assert job_tracker.running_jobs["job123"]["job_name"] == "TestJob"
+        assert "start_time" in job_tracker.running_jobs["job123"]
+
+    def test_when_job_already_running_on_job_submitted_should_not_update(self):
+        """Test when a job is submitted but already in running list."""
+        job_tracker = JobTracker()
+        job_tracker.running_jobs["job123"] = {"start_time": datetime.now()}
+
+        event = Mock(spec=JobSubmissionEvent)
+        event.job_id = "job123"
+        event.scheduled_run_times = [datetime.now()]
+        event.jobstore = "default"
+
+        original_info = job_tracker.running_jobs["job123"]
+        job_tracker.on_job_submitted(event)
+
+        # Should not update the existing entry
+        assert job_tracker.running_jobs["job123"] == original_info
+
+    def test_when_no_scheduled_time_on_job_submitted_should_use_current_time(self):
+        """Test when job has no scheduled run times."""
+        job_tracker = JobTracker()
+        event = Mock(spec=JobSubmissionEvent)
+        event.job_id = "job123"
+        event.scheduled_run_times = []
+        event.jobstore = "default"
+
+        job_tracker.on_job_submitted(event)
+
+        assert "job123" in job_tracker.running_jobs
+        assert "scheduled_run_time" in job_tracker.running_jobs["job123"]
```

Comment:
This does not check that the current time is used...at least you could check that `job_tracker.running_jobs["job123"]["scheduled_run_time"] is not None`